iteration/iteration_test.go

```diff
@@ -204,62 +174,49 @@ func (s *TestIterationRepository) TestListIterationBySpace() {
 
 		i := iteration.Iteration{
 			Name:       name,
-			SpaceID:    spaceInstance.ID,
+			SpaceID:    fxt.Spaces[0].ID,
 			StartAt:    &start,
 			EndAt:      &end,
 			UserActive: &userActive,
 		}
 		e := repo.Create(context.Background(), &i)
 		require.Nil(t, e)
 	}
-	// create another space and add iteration to another space
-	anotherSpace := space.Space{
-		Name:    "Space 2",
-		OwnerId: s.testIdentity.ID,
-	}
-	anotherSpaceCreated, err := repoSpace.Create(context.Background(), &anotherSpace)
-	assert.Nil(t, err)
+	// add iteration to the second space from our fixture
 	e := repo.Create(context.Background(), &iteration.Iteration{
 		Name:       "Other Spring #2",
-		SpaceID:    anotherSpaceCreated.ID,
+		SpaceID:    fxt.Spaces[1].ID,
 		UserActive: &userActive,
 	})
 	require.Nil(t, e)
 
-	its, err := repo.List(context.Background(), spaceInstance.ID)
+	its, err := repo.List(context.Background(), fxt.Spaces[0].ID)
 	assert.Nil(t, err)
 	assert.Len(t, its, 3)
 }
 
 func (s *TestIterationRepository) TestUpdateIteration() {
 	t := s.T()
 	resource.Require(t, resource.Database)
-
-	userActive := false
 	repo := iteration.NewIterationRepository(s.DB)
 
 	start := time.Now()
 	end := start.Add(time.Hour * (24 * 8 * 3))
-	name := "Sprint #24"
-
-	newSpace := space.Space{
-		Name:    "Space 1",
-		OwnerId: s.testIdentity.ID,
-	}
-	repoSpace := space.NewRepository(s.DB)
-	space, err := repoSpace.Create(context.Background(), &newSpace)
-	assert.Nil(t, err)
 
-	i := iteration.Iteration{
-		Name:       name,
-		SpaceID:    space.ID,
-		StartAt:    &start,
-		EndAt:      &end,
-		UserActive: &userActive,
-	}
-	// creates an iteration
-	repo.Create(context.Background(), &i)
-	require.NotEqual(t, uuid.Nil, i.ID, "Iteration was not created, ID nil")
+	fxt := tf.NewTestFixture(s.T(), s.DB,
+		tf.Iterations(1,
+			tf.UserActive(false),
```

Comment:
I did not notice that `tf.UserActive(bool)` function before. Why is it not part of the customize function ?

Reply:
or is it meant to be applied to all iterations ? anyways, I'd rather see it in the function below than in a separate function (IMHO)

Reply:
I just wanted to introduce a specialized / reusable customize function. So that you don't have to write the longish function. The UserActive is applied to all iterations, yes.

Reply:
I'm not entirely convinced by this, honestly: it does not make the code simpler to read.